File: src/test/java/morsecode/MorseCodeTest.java
package morsecode;

import org.junit.jupiter.api.Test;

import static java.util.Set.of;
import static morsecode.MorseCode.decode;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class MorseCodeTest {

    @Test
    public void testDecodeMorseCode() {
        assertEquals(of("hello"), decode("......_..._..___"));
        assertEquals(of("world"), decode("._____._.._.._.."));
    }

    @Test
    public void testDecodeMorseCodeWithMultipleCombinations() {
        assertEquals(of("lever", "river"), decode("._......_.._."));
    }
}

File: src/test/kotlin/morsecode/MorseCodeTest.kt
package morsecode

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import morsecode.MorseCode.decode

class MorseCodeTest {
    @Test
    fun `test decode morse code`() {
        assertEquals(setOf("hello"), decode("......_..._..___"))
        assertEquals(setOf("world"), decode("._____._.._.._.."))
    }

    @Test
    fun `test decode morse code with multiple combinations`() {
        assertEquals(setOf("lever", "river"), decode("._......_.._."))
    }
}
